src/CoercibleUtils.hs

```diff
@@ -1,8 +1,58 @@
-module CoercibleUtils where
+{- |
+Utility functions for Coercible types.
+Primarily pulled from the package `newtype-generics`, and
+based on Conor McBride's Epigram work, but
+generalised to work over anything `Coercible`.
 
-import Data.Coerce
+>>> ala Sum foldMap [1,2,3,4]
+10
+
+>>> ala Endo foldMap [(+1), (+2), (subtract 1), (*2)] 3
+8
+
+>>> under2 Min (<>) 2 1
+1
+
+>>> over All not (All False)
+All {getAll = True)
+
+Users might also find the GHC plugin <https://github.com/mpickering/hashtag-coerce hashtag-coerce> useful in tandem with this library.
```

Comment:
Have you tested the plugin? Is it easy to use and are the suggestions useful? I'm a bit wary of recommending something  that might not work well.